elements/pf-progress/pf-progress.ts

```diff
@@ -120,83 +122,102 @@ const ICONS = new Map(Object.entries({
 export class PfProgress extends LitElement {
   static readonly styles = [styles];
 
+  static readonly formAssociated = true;
+
+  #calculatedPercentage = 0;
+
+  #logger = new Logger(this);
+
   @property({ reflect: true, type: Number })
     value = 0;
 
-  @property({ reflect: true })
+  @property()
     title = '';
 
-  @property({ reflect: true })
-    label = '';
+  @property({ attribute: 'aria-label' })
+    ariaLabel = '';
+
+  @property({ type: Number })
+    max = 100;
+
+  @property({ type: Number })
+    min = 0;
 
-  @property({ reflect: true })
+  @property()
     size: 'sm' | 'lg' | '' = '';
 
   @property({ reflect: true, attribute: 'measure-location' })
     measureLocation: '' | 'outside' | 'inside' | 'none' = '';
 
-  @property({ reflect: true })
-    markdown: 'html' | 'progress' | 'meter' = 'html';
-
-  @property({ reflect: true })
+  @property()
     variant: '' | 'success' | 'danger' | 'warning' = '';
 
-  @query('#bar')
-    bar!: HTMLElement;
+  @query('progress')
+    progressElement!: HTMLProgressElement;
 
   #showInsideStatus() {
     return this.measureLocation === 'inside';
   }
 
-  async #updateAccessibility() {
-    !!await this.updateComplete;
-
-    const { bar, value } = this;
-    bar.setAttribute('aria-valuenow', `${value}`);
-    if (this.title.length > 0 && this.bar !== null) {
-      this.bar.setAttribute('aria-labelledby', 'title');
-    } else if (this.bar !== null) {
-      this.bar.setAttribute('aria-label', this.label);
+  #calculatePercentage() {
+    const { value, max, min } = this;
+    if (value === null ) {
+      this.#logger.warn(`A progress element requires a value attribute.`);
+      return;
     }
+    const percentage = (value - min) / (max - min);
+    this.#calculatedPercentage = Math.round(percentage * 100);
```

Comment:
let's either move all this to scoped consts in the render method, or have this private method return it's value rather than mutating private state.

the logger call, since it's a side effect should move to `willUpdate`